tests/transforms/test_classical_jacobian.py

```diff
@@ -121,13 +129,14 @@ def test_autograd_without_argnum(circuit, args, expected_jac, diff_method):
     jac = classical_jacobian(qnode)(*args)
 
     # NOTE: We use stacking to replicate qml.jacobian behaviour for scalar-only inputs
-    if all((np.isscalar(arg) for arg in args)):
+    arg_shapes = [qml.math.shape(arg) for arg in args]
+    if len(args) == 1:
+        expected_jac = expected_jac[0]
+    if all(sh == arg_shapes[0] for sh in arg_shapes[1:]):
         expected_jac = qml.math.stack(expected_jac).T
         assert np.allclose(jac, expected_jac)
     else:
         # For a single argument, the Jacobian is unpacked
```

Comment:
Move this comment up:
```suggestion
    else:
```